rules/detekt/src/test/kotlin/io/nlopez/compose/rules/detekt/LambdaParameterInRestartableEffectCheckTest.kt

```diff
@@ -78,4 +78,65 @@ class LambdaParameterInRestartableEffectCheckTest {
         val errors = rule.lint(code)
         assertThat(errors).isEmpty()
     }
+
+    @Test
+    fun `error out when detecting a lambda named onDispose used in a non-DisposableEffect`() {
+        @Language("kotlin")
+        val code =
+            """
+                @Composable
+                fun Something(onDispose: () -> Unit) {
+                    LaunchedEffect(Unit) {
+                        onDispose()
+                    }
+                }
+                @Composable
+                fun Something(onDispose: () -> Unit) {
+                    DisposableEffect(Unit) {
+                        onDispose(onDispose)
+                    }
+                }
+
+                // TODO ideally these would also be caught, but may require type resolution
```

Comment:
Yea, that's fine, this would catch most of cases anyway so it's ok it doesn't work for these, there's no good way of dealing with it anyway.